src/settings.cpp

```diff
@@ -764,24 +766,48 @@ bool Settings::getFlagStrNoEx(const std::string &name, u32 &val,
 bool Settings::setEntry(const std::string &name, const void *data,
 	bool set_group, bool set_default)
 {
-	Settings *old_group = NULL;
 
+	// Return if setting name is invalid
 	if (!checkNameValid(name))
 		return false;
-	if (!set_group && !checkValueValid(*(const std::string *)data))
+
+	std::string value;
+	if (!set_group)
+		value = *(const std::string *)data;
+
+	// Return if value is invalid and we're not dealing with a setting group
+	if (!set_group && !checkValueValid(value))
 		return false;
 
+	std::cout << "\nSettings::setEntry(" << name << ")" << std::endl;
+
+	// Write setting only if it has been modified. Return true to retain compatibility
+	if (!set_default && !set_group) {
+		SettingsEntry old_entry;
+		if (getEntryNoEx(name, old_entry) && old_entry.value == value)
+			return true;
+	}
+
+	if (!set_default)
+		std::cout << "\tSetting modified. new=" << value << "\n" << std::endl;
+	else
+		std::cout << "\tSetting default. default=" << value << "\n" << std::endl;
+
+	Settings *old_group = nullptr;
 	{
 		MutexAutoLock lock(m_mutex);
 
 		SettingsEntry &entry = set_default ? m_defaults[name] : m_settings[name];
-		old_group = entry.group;
 
-		entry.value    = set_group ? "" : *(const std::string *)data;
-		entry.group    = set_group ? *(Settings **)data : NULL;
+		if (set_group) {
+			old_group = entry.group;
+			entry.group = *(Settings **)data;
+		} else {
+			entry.value = value;
```

Comment:
Missing garbage clean-up.
`entry.group` must be freed and set to `nullptr`, and in the upper scope: `entry.value = "";`

I'd prefer if you could go back to the previous notation which clearly stated that both variables must be updated.

Reply:
Ok, will revert to previous notation.